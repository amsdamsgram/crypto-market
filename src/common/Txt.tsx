import React from "react";
import styled from "styled-components/native";

import Theme from "../Theme";

interface IProps {
  children?: any;
  style?: any;
}

const StyledText = styled.Text`
  color: ${Theme.colors.text};
  font-size: ${Theme.globalStyles.fontSize}px;
`;

const Txt = (props: IProps) => {
  return (
    <StyledText {...props} style={props.style}>
      {props.children}
    </StyledText>
  );
};

Txt.defaultProps = {
  children: null,
  style: null
};

export default Txt;
