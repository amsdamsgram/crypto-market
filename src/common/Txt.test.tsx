import React from "react";
import renderer from "react-test-renderer";

import Txt from "./Txt";

describe("Txt", () => {
  test("should match snapshot", () => {
    const tree = renderer.create(<Txt>Hello</Txt>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
