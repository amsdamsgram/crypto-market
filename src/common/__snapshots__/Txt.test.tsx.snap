// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Txt should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#414141",
        "fontSize": 14,
      },
      null,
    ]
  }
>
  Hello
</Text>
`;
